solutions/kubernetes/ceph-binary-build.sh

```diff
@@ -0,0 +1,299 @@
+#!/bin/bash
+#
+# Copyright (c) 2022 Seagate Technology LLC and/or its Affiliates
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+######################################################################
+# This script is specific for the reserved HW by RE team.
+# To use on other HW/VM please install docker and change
+# container volume $MOUNT_LOCATION for build os container.
+######################################################################
+
+source functions.sh
+
+MOUNT_LOCATION="/var/log/ceph-build"
+
+ACTION="$1"
+if [ -z "$ACTION" ]; then
+    echo "ERROR : No option provided"
+    usage
+    exit 1
+fi
+
+function usage() {
+    cat << HEREDOC
+Usage : $0 [--build-ubuntu, --build-centos, --build-rockylinux]
+where,
+    --build-ubuntu - Build Ubuntu binary packages (*.debs).
+    --build-centos  - Build CentOS binary packages (*.rpms).
+    --build-rockylinux - Build Rocky Linux binary packages (*.rpms).
+HEREDOC
+}
+
+function check_params() {
+    if [ -z "$CEPH_REPO" ]; then echo "CEPH_REPO not provided. Using default: ceph/ceph ";CEPH_REPO="ceph/ceph"; fi
+    if [ -z "$CEPH_BRANCH" ]; then echo "CEPH_BRANCH not provided. Using default: quincy";CEPH_BRANCH="quincy"; fi
+
+   echo -e "\n\n########################################################################"
+   echo -e "# CEPH_REPO         : $CEPH_REPO                  "
+   echo -e "# CEPH_BRANCH       : $CEPH_BRANCH                "
+   echo -e "#########################################################################"
+}
+
+function prereq() {
+    add_primary_separator "\t\tPreparing Build node"
+
+    # Verify docker installation:
+    add_secondary_separator "Verify docker installation"
+    if ! which docker; then
+        add_secondary_separator "Please install Docker on Build Node Agent"
+        exit 1
+    fi
+}
+
+function build_ubuntu() {
+    add_primary_separator "Building ubuntu binary packages"
+
+    add_secondary_separator "Verfiy/pull Ubuntu docker image"
+    if [[ $(docker images | tail -n+2 | awk '{ print $1":"$2 }' | grep ubuntu) != "ubuntu:latest" ]]; then
```

Comment:
You can use `reference` option in `docker images` to check if the required image is already available. Refer - https://docs.docker.com/engine/reference/commandline/images/